components/carousel/script.js

```diff
@@ -207,27 +211,19 @@ class Carousel extends BaseComponent {
      */
     prevStepIndex(stepSize) {
         const itemsLength = this.items.length;
-        return (this.current - stepSize + itemsLength) % itemsLength;
+        return (this.currentItemIndex - stepSize + itemsLength) % itemsLength;
     }
 
     /**
-     * Check if the up nav arrow should be visible - if navigating up is possible.
-     * If not - hide it.
+     * Check if a nav arrow based on a given direction should be
+     * visible - if navigating in that direction is possible
+     * @param {number} direction - -1  for left and 1 for right
+     * @param {number} nextIndex - the index of the next item
      * @returns {undefined}
      */
-    shouldShowUpNavArrow() {
-        if (this.overflows(this.items.length, this.nextStepIndex(this.navArrowStepSize))) return this.hideArrow(1);
-        this.showArrow(1);
-    }
-
-    /**
-     * Check if the down nav arrow should be visible - if navigating down is possible.
-     * If not - hide it.
-     * @returns {undefined}
-     */
-    shouldShowDownNavArrow() {
-        if (this.overflows(this.items.length, this.prevStepIndex(this.navArrowStepSize))) return this.hideArrow(-1);
-        this.showArrow(-1);
+    shouldShowNavArrow(direction, nextIndex) {
+        if (this.overflows(nextIndex)) return this.hideArrow(direction);
+        this.showArrow(direction);

```

Comment:
Instead of passing the `nextIndex` why not retrieve it directly in the method body? This will shorten the method call.

```suggestion
    shouldShowNavArrow(direction) {
        const nextIndex = direction === DIRECTIONS.RIGHT ? this.nextStepIndex(this.navArrowStepSize) : this.prevStepIndex(this.navArrowStepSize);
        if (this.overflows(nextIndex)) return this.hideArrow(direction);
        this.showArrow(direction);
```